### ISSUE TITLE
Feature request: Show notifications of internal osm messages to the StreetComplete user

### ISSUE THREAD

**Issue by sun-geo** [westnordost/StreetComplete#1690]
Does the current version push internal osm messages from other osm-members to the StreetComplete app user (before and even while an edit session)?
Like messages which are send via system https://www.openstreetmap.org/message/new/westnordost

The background of this question is, can i reach a 100% SteetComplete-editor-user via the osm-internal message system beside any email app tools? If not, please implement such an option to do so.

I think the iD- and JOSM-editor are showing users, when new internal osm messages are received. 

**Use case**

Talk to other osm-users about their database changes. 

**Proposed Solution**

Add a message-viewer or even chat-option for StreetComplete.

edit: typo

**Comment by westnordost** [comment]
No, StreetComplete does no such thing at the moment. Perhaps you can reformulate the issue title and description  to make clear that this is actually a feature request rather than a question.

Could you also elaborate on why an email notification about new messages is not enough? After all, I think neither josm nor iD show changeset discussions even though they may actually be more important.

Am 3. Januar 2020 00:28:53 MEZ schrieb sun-geo <notifications@github.com>:
>Does the current version push internal osm messages from other
>osm-members to the StreetComplete app user (before and even while an
>edit session)?
>Like messages which are send via system
>https://www.openstreetmap.org/message/new/westnordost
>
>The background of this question is, can i reach a 100%
>SteetComplete-editor-user via the osm-internal message system beside
>any email app tools? If not, please implement such an option to do so.
>
>I thing the iD- and JOSM-editor are showing users, when new internal
>osm messages are received. 
>
>
>
>**Use case**
>
>
>Talk to after osm-users about their database changes. 
>
>**Proposed Solution**
>
>
>Add a message-viewer or even chat-option for StreetComplete.

**Comment by matkoniecz** [comment]
JOSM is showing notifications about unread messages, iD is on osm page that has build-in display of this info.

Changeset comments are not shown in this way due to technical limitations.

**Comment by matkoniecz** [comment]
> Could you also elaborate on why an email notification about new messages is not enough?

Many people are not reacting to emails or are checking it rarely. It is common that people have thousands of unread messages in inboxes. This way people are more likely to notice that PM arrived.

And even in case of regularly checking email unread message notification allows to respond with reduced lag, in some cases it allowed me to exchange several messages within one hour. With "check email once a day" method it would take weeks.

**Comment by sun-geo** [comment]
> Perhaps you can reformulate the issue title

done

> Could you also elaborate on why an email notification about new messages is not enough?

See also the answer from matkoniecz & for me is checking emails very low priority (maybe loading emails in average only once per month), so i or other StreetComplete user (or especially osm user which use StreetComplete as exclusive editor) would see the notification of StreetComplete app earlier (e.g. immediately when they open Streetcomplete, once the feature is implemented), when i/they using the app in a higher frequency than once per month...

> changeset discussions comments

This is a different topic. Btw these (public messages) are also low priority for me, because my experience is, people tend to reply faster (if they get a notice ;-) ) and more open and friendly to internal messages.

**Comment by Atrate** [comment]
This feature could also manifest its usefulness because of the achievement system, which has the possibility of creating an influx of more casual mappers that don't really know much about OSM.

**Comment by westnordost** [comment]
https://www.westnordost.de/misc/streetcomplete-new-message.m4v

**Comment by sun-geo** [comment]
@westnordost 

Did test this feature today & works very well!

Thanks for implementation

:-)
